### PR TITLE
Awaiting channel

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

File: driver/src/main/scala/api/cursor.scala [hunk 1]
```diff
@@ -687,11 +615,18 @@ object DefaultCursor {
...
         logger.debug(s"[$logCat] Clean up $cursorID, sending KillCursors")
 
-        val killReq = RequestMaker(
-          KillCursors(Set(cursorID)),
-          readPreference = preference)
+        val killReq = RequestMakerExpectingResponse(
+          RequestMaker(KillCursors(Set(cursorID)), readPreference = preference),
+          false)
+
+        import ExecutionContext.Implicits.global
```
Comment (jtjeferreira) [inline driver/src/main/scala/api/cursor.scala]: ExecutionContext.Implicits.global

Comment (cchantep) [reply driver/src/main/scala/api/cursor.scala]: That's WIP

Comment (jtjeferreira) [reply driver/src/main/scala/api/cursor.scala]: :+1:

**Comment by eugeneatnezasa** [comment]
Strange behavior if I set nbChannelsPerNode=1, occupy this channel by slow query and then send several fast queries:
several fast queries fail (No active channel ...), several executed successfully
And everything happens after slow query completed

**Comment by cchantep** [comment]
Which is coherent, if no channel is available in the pool, it fail. Alternative would be to queue request.

**Comment by cchantep** [comment]
@eugeneatnezasa unless you've remove the failover, this is used to retry (and possibly succeed).

**Comment by eugeneatnezasa** [comment]
sorry, it was a mistake with approval (mixed up multiple github windows)

**Comment by eugeneatnezasa** [comment]
@cchantep I approved twice this PR by mistake (wrong window). Just notifying

**Comment by cchantep** [comment]
@eugeneatnezasa You're free to approve if it's ok for you ;-)

**Pull request merged**